alphalens/tears.py

```diff
@@ -393,6 +395,24 @@ def create_returns_tear_sheet(factor_data,
                                            ylim_percentiles=(5, 95),
                                            ax=ax_quantile_returns_bar_by_group)
 
+    ff_factors = ep.utils.load_portfolio_risk_factors() \
+        .drop(['RF'], axis='columns')
+    hierarchy = OrderedDict([
+        # not typos! These are the Fama-French factors names from Dartmouth
+        ('Market', ['Mkt-RF']),
+        ('Style', ['SMB', 'HML', 'Mom'])
+    ])
+    if factor_returns.index.tzinfo is None:
+        factor_returns.index = factor_returns.index.tz_localize('UTC')
+    else:
+        factor_returns.tz_convert('UTC')
+    for i, period in enumerate(factor_returns.columns.values):
+        returns_decomposition = \
+            perf.decompose_returns(factor_returns.iloc[:, i],
```

Comment:
I have just realized that passing 'factor_returns' to `perf.decompose_returns` works only in the very specific scenario (thought very common and probably the most common scenario), that is when the factor is:
- computed on a daily basis
- the period is 1 day
- the entry/exit prices for the factor are market open